lib/resources/oracledb_session.rb

```diff
@@ -49,19 +49,23 @@ def query(q)
 
       p = nil
       # use sqlplus if sqlcl is not available
-      if inspec.command(@sqlcl_bin).exist?
+      if @sqlcl_bin and inspec.command(@sqlcl_bin).exist?
         bin = @sqlcl_bin
         opts = "set sqlformat csv\nSET FEEDBACK OFF"
         p = :parse_csv_result
       else
         bin = @sqlplus_bin
-        opts = "SET MARKUP HTML ON\nSET FEEDBACK OFF"
+        opts = "SET MARKUP HTML ON\nSET PAGESIZE 32000\nSET FEEDBACK OFF"
         p = :parse_html_result
       end
 
       query = verify_query(escaped_query)
       query += ';' unless query.end_with?(';')
-      command = %{echo "#{opts}\n#{query}\nEXIT" | #{bin} "#{@user}"/"#{@password}"@#{@host}:#{@port}/#{@service}}
+      if /as (sysdba|sysoper|sysasm)/i === @password
+        command = %{su - #{user} -c "env ORACLE_SID=#{@service} #{bin} / #{@password} <<EOC\n#{opts}\n#{query}\nEXIT\nEOC"}
```

Comment:
I'm not familiar with oracle tools, so I searched a bit as this seemed like an odd construction.  I found [this](https://asktom.oracle.com/pls/apex/f?p=100:11:0::::P11_QUESTION_ID:409364600346264472); it makes me think that the two branches of this conditional might rely on different credential resolution systems; that would seem to be an unexpected side effect.

Reply:
In article you mentioned they said:
/ as sysdba - connect to LOCAL instance as specified in the environment variables ORACLE_HOME + ORACLE_SID

sys/a@fstest as sysdba - connect OVER A NETWORK to a remote database, using the credentials sys/a

I'm using the first way - connect to the local instance. That is why I need to do su to some user and provide ORACLE_SID in environment. In this case I don't need to provide any sensitive data, like real sysdba credentials. But this is possible only as a particular user (oracle, grid, etc).

Reply:
Yes I agree, oracle has been always very hard to test. I had to resort to login shell to get the environmental vars sourced. This might fail in the case the `oracle` user has no sudo access and cannot use `su` we might need to handle such conditions too.